gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java

```diff
@@ -471,44 +434,42 @@ public void deploy(FsDeploymentConfig deploymentConfig) throws IOException {
 
     Path hdfsNewVersionPath = new Path(hdfsconfigStoreRoot, deploymentConfig.getNewVersion());
 
-    if (fs.exists(hdfsNewVersionPath)) {
-      log.warn(String.format("Version %s already exits at %s. Can not overwrite an existing version.",
-          deploymentConfig.getNewVersion(), hdfsNewVersionPath));
-    }
+    if (!fs.exists(hdfsNewVersionPath)) {
 
-    Set<ConfigStream> confStreams = deploymentConfig.getDeployableConfigSource().getConfigStreams();
+      fs.mkdirs(hdfsNewVersionPath, deploymentConfig.getStorePermissions());
 
-    for (ConfigStream confStream : confStreams) {
-      String confAtPath = confStream.getConfigPath();
+      Set<ConfigStream> confStreams = deploymentConfig.getDeployableConfigSource().getConfigStreams();
 
-      log.info("Copying resource at : " + confAtPath);
+      for (ConfigStream confStream : confStreams) {
+        String confAtPath = confStream.getConfigPath();
 
-      Path hdsfConfPath = new Path(hdfsNewVersionPath, confAtPath);
+        log.info("Copying resource at : " + confAtPath);
 
-      if (!fs.exists(hdsfConfPath.getParent())) {
-        fs.mkdirs(hdsfConfPath.getParent());
-      }
+        Path hdsfConfPath = new Path(hdfsNewVersionPath, confAtPath);
+
+        if (!fs.exists(hdsfConfPath.getParent())) {
+          fs.mkdirs(hdsfConfPath.getParent());
+        }
 
-      // If an empty directory needs to created it may not have a stream.
-      if (confStream.getInputStream().isPresent()) {
-        // Read the resource as a stream from the classpath and write it to HDFS
-        try (SeekableFSInputStream inputStream = new SeekableFSInputStream(confStream.getInputStream().get());
-            FSDataOutputStream os = this.fs.create(hdsfConfPath, false)) {
-          StreamUtils.copy(inputStream, os);
+        // If an empty directory needs to created it may not have a stream.
+        if (confStream.getInputStream().isPresent()) {
+          // Read the resource as a stream from the classpath and write it to HDFS
+          try (SeekableFSInputStream inputStream = new SeekableFSInputStream(confStream.getInputStream().get());
+              FSDataOutputStream os = this.fs.create(hdsfConfPath, false)) {
```

Comment:
Can you add an explicit check to see if the config file already exists, that way we can control the error message and make it more descriptive than `FileAlreadyExistsException`